dds/DCPS/ValueReader.h

```diff
@@ -100,7 +100,9 @@ class ListEnumHelper : public EnumHelper {
   const Pair* pairs_;
 };
 
-struct ValueReader {
+class OpenDDS_Dcps_Export ValueReader {
+public:
```

Comment:
Does it have a class invariant?  Is anything private?  If neither is true, isn't `struct` clear and sufficient?

Reply:
I personally prefer `class`

Reply:
Not at all a requirement... just to give more background here, we should try to follow C++ Core Guidelines when there isn't a project-specific guideline.  In this case there may not be a clear consensus.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c122-use-abstract-classes-as-interfaces-when-complete-separation-of-interface-and-implementation-is-needed

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c2-use-class-if-the-class-has-an-invariant-use-struct-if-the-data-members-can-vary-independently

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c8-use-class-rather-than-struct-if-any-member-is-non-public